### PR TITLE
Fixed #32560 -- Fixed test runner with --pdb and --buffer on fail/error.

### PR DESCRIPTION
**Pull request by adamchainz**
ticket-32560

### CODE REVIEW

**Comment by adamchainz** [comment]
I tried writing a test, came up with this for `DiscoverRunnerTests`:

```
def test_buffer_mode_pdb(self):
        runner = DiscoverRunner(pdb=True, buffer=True, verbose=0)
        with captured_stdin() as stdin, captured_stdout() as stdout, captured_stderr() as stderr:
            stdin.write("continue\n")
            suite = runner.build_suite([
                'test_runner_apps.buffer.tests_buffer.WriteToStdoutStderrTestCase.test_fail',
            ])
            runner.run_suite(suite)
        self.assertIn('Write to stderr.', stderr.getvalue())
        self.assertIn('Write to stdout.', stdout.getvalue())
```

Unfortunately it doesn't expose the issue proper, since it passes *before* the test run. We need something more like "expect" / `pexpect` to start the test runner and assert that the test runner has output the captured stdout and pdb prompt before runing `continue`, but that would be a bit complicated/slow to me.

**Comment by felixxm** [comment]
@adamchainz Thanks :+1: Can you add release notes? (3.1.8)

**Comment by adamchainz** [comment]
Added!

**Comment by felixxm** [comment]
@adamchainz Thanks :+1: I pushed edits to release notes.

**Review by adamchainz — COMMENTED** [review]
LGTM

File: docs/releases/3.1.8.txt [hunk 1]
```diff
@@ -9,4 +9,6 @@ Django 3.1.8 fixes several bugs in 3.1.7.
 Bugfixes
 ========
 
-* ...
+* Fixed a bug in Django 3.1 where the output was hidden on test error or
```
Comment (felixxm) [inline docs/releases/3.1.8.txt:12]:
```suggestion
* Fixed a bug in Django 3.1 where the output was hidden on a test error or
```
I missed `a`.

**Pull request merged**